src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseCodeReviewFeedbackService.java

```diff
@@ -0,0 +1,239 @@
+        if (this.athenaFeedbackSuggestionsService != null) {
+            this.checkRateLimitOrThrow(participation);
+            CompletableFuture.runAsync(() -> this.generateAutomaticNonGradedFeedback(exerciseId, participation, programmingExercise));
+            return participation;
+        }
+        else {
+            log.debug("tutor is responsible to process feedback request: {}", exerciseId);
+            groupNotificationService.notifyTutorGroupAboutNewFeedbackRequest(programmingExercise);
+            return setIndividualDueDateAndLockRepository(participation, programmingExercise, true);
+        }
+    }
+
+    /**
+     * Generates automatic non-graded feedback for a programming exercise submission.
+     * This method leverages the Athena service to generate feedback based on the latest submission.
+     *
+     * @param exerciseId          the id of the programming exercise.
+     * @param participation       the student participation associated with the exercise.
+     * @param programmingExercise the programming exercise object.
+     */
+    public void generateAutomaticNonGradedFeedback(Long exerciseId, ProgrammingExerciseStudentParticipation participation, ProgrammingExercise programmingExercise) {
+        log.debug("Using athena to generate feedback request: {}", exerciseId);
+
+        // athena takes over the control here
+        var submissionOptional = programmingExerciseParticipationService.findProgrammingExerciseParticipationWithLatestSubmissionAndResult(participation.getId())
+                .findLatestSubmission();
+        if (submissionOptional.isEmpty()) {
+            throw new BadRequestAlertException("No legal submissions found", "submission", "noSubmission");
+        }
+        var submission = submissionOptional.get();
+
+        // save result and transmit it over websockets to notify the client about the status
+        var automaticResult = this.submissionService.saveNewEmptyResult(submission);
+        automaticResult.setAssessmentType(AssessmentType.AUTOMATIC_ATHENA);
+        automaticResult.setRated(false);
+        automaticResult.setScore(100.0); // requests allowed only if all autotests pass, => 100
+        automaticResult.setSuccessful(null);
+        automaticResult.setCompletionDate(ZonedDateTime.now().plusMinutes(5)); // we do not want to show dates without a completion date, but we want the students to know their
+                                                                               // feedback request is in work
+        automaticResult = this.resultRepository.save(automaticResult);
+
+        try {
+
+            setIndividualDueDateAndLockRepository(participation, programmingExercise, false);
+            this.programmingMessagingService.notifyUserAboutNewResult(automaticResult, participation);
+            // now the client should be able to see new result
+
+            log.debug("Submission id: {}", submission.getId());
```

Comment:
This log statement is not very descriptive. Please improve it so that it actually states what is happening.